### PR TITLE
virtctl: Allow size to be optional when creating VMs

### PR DESCRIPTION
**Pull request by 0xFelix**
### What this PR does

Allow to omit size when creating a VM and using --volume-import to clone a PVC or a VolumeSnapshot.

Before this PR:

Specifying size when creating a VM and using --volume-import to clone a PVC or a VolumeSnapshot was always required.

After this PR:

Specifying size when creating a VM and using --volume-import to clone a PVC or a VolumeSnapshot is optional now.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 
-

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
virtctl: Specifying size when creating a VM and using --volume-import to clone a PVC or a VolumeSnapshot is optional now
```

### CODE REVIEW

**Comment by alicefr** [comment]
@0xFelix a super nit, if you can split the ErrorNotFound type in a separate commit

**Comment by alicefr** [comment]
The code looks good to me! Thanks for the work
/approve

File: pkg/virtctl/create/vm/vm.go [hunk 1]
```diff
@@ -247,6 +248,11 @@ var volumeImportOptions = map[string]volumeImportFn{
 	snapshot: withVolumeSourceSnapshot,
 }
 
+var volumeImportSizeOptional = map[string]bool{
+	pvc:      true,
+	snapshot: true,
+}
```
Comment (fossedihelm) [inline pkg/virtctl/create/vm/vm.go:254]: nit: I'd prefer
```suggestion
var volumeImportSizeMandatory = map[string]bool{
	pvc:      false,
	snapshot: false,
}
```
and save a **not** operation below.
But maybe it is my brain that starts burning when it see more than one negation :)

**Review by fossedihelm — APPROVED** [review]
Thanks @0xFelix!
One super nit that you can ignore.
Feel free to unhold
/hold

**Comment by 0xFelix** [comment]
/hold cancel

Discussed this offline, will keep as is.

**Pull request merged**